src/tests/unit/tests/common/components/issue-filing-button.test.tsx

```diff
@@ -0,0 +1,133 @@
+
+describe('IssueFilingButtonTest', () => {
+    let environmentInfoProviderMock: IMock<EnvironmentInfoProvider>;
+    let bugFilingServiceProviderMock: IMock<BugFilingServiceProvider>;
+    let bugActionMessageCreatorMock: IMock<BugActionMessageCreator>;
+    let userConfigurationStoreData: UserConfigurationStoreData;
+    const testKey: string = 'test';
+    let testBugService: BugFilingService;
+
+    beforeEach(() => {
+        testBugService = {
+            key: testKey,
+            displayName: 'TEST',
+            settingsForm: NamedSFC('testForm', props => <>Hello World</>),
+            isSettingsValid: () => true,
+            buildStoreData: testField => {
+                return { testField };
+            },
+            getSettingsFromStoreData: data => data[testKey],
+            createBugFilingUrl: () => 'test url',
+        };
+        userConfigurationStoreData = {
+            bugService: testKey,
+            bugServicePropertiesMap: {},
+        } as UserConfigurationStoreData;
+        environmentInfoProviderMock = Mock.ofType(EnvironmentInfoProvider);
+        bugFilingServiceProviderMock = Mock.ofType(BugFilingServiceProvider);
+        bugActionMessageCreatorMock = Mock.ofType(BugActionMessageCreator);
+        environmentInfoProviderMock
+            .setup(envp => envp.getEnvironmentInfo())
+            .returns(() => {
+                return {
+                    extensionVersion: '1',
+                    axeCoreVersion: '2',
+                    browserSpec: 'spec',
+                };
+            })
+            .verifiable();
+        bugFilingServiceProviderMock
+            .setup(bp => bp.forKey(testKey))
+            .returns(() => testBugService)
+            .verifiable();
+    });
+
+    test.each([true, false])('render: isSettingsValid: %s', isSettingsValid => {
+        testBugService.isSettingsValid = () => isSettingsValid;
+        const props: IssueFilingButtonProps = {
+            deps: {
+                bugActionMessageCreator: bugActionMessageCreatorMock.object,
+                environmentInfoProvider: environmentInfoProviderMock.object,
+                bugFilingServiceProvider: bugFilingServiceProviderMock.object,
+            } as IssueFilingButtonDeps,
+            issueDetailsData: {
+                pageTitle: 'pageTitle',
+                pageUrl: 'http://pageUrl',
+                ruleResult: null,
+            },
+            userConfigurationStoreData: userConfigurationStoreData,
+        };
+        const wrapper = shallow(<IssueFilingButton {...props} />);
+        expect(wrapper.debug()).toMatchSnapshot();
+
+        environmentInfoProviderMock.verifyAll();
+        bugFilingServiceProviderMock.verifyAll();
+    });
+
+    test('onclick: valid settings, file bug', () => {
+        bugActionMessageCreatorMock
+            .setup(messageCreator => messageCreator.trackFileIssueClick(It.isAny(), testKey as any))
```

Comment:
can you make the assertion of the first parameter more robust? `It.isAny()` doesn't let you validate much and we already have a `EventStubFactory` ready to use on our tests.

Reply:
we are using simulate('click') to test on click, which we cannot pass an event stub.

Reply:
We can. Look for examples on our own code.

Reply:
Done.